### DISCUSSION TITLE
I need help with arrays and struct

### DISCUSSION THREAD

**Discussion by yashjh** [Cyfrin/foundry-full-course-cu#3045] [Q&A]
I am getting confused with syntaxes
why the declared favoritenumber and name isn't used anywhere in code but used with underscore why ??

**Comment by heheboii11** [comment]
Can you just share the context little bit, may be share the part of the code you are facing error with ?

**Reply by yashjh** [reply]
yes afcause,
![Screenshot 2024](https://github.com/user-attachments/assets/14580c62-d9bc-4788-a373-08d7b7e0cc97)
just explain me that uint256 favoriteNumber  string name are declared in struct person but not used as same name in fuction but used with underscore names why ??

**Reply by chainflash** [reply]
bro,  ***_favoriteNumber*** is different from ***favoriteNumber***, for example, we can name it anything like ***_random***.
In the function, we define a new variable of type uint256, This new variable is used for assignment.
Similarly, ***name*** and ***_name*** are also like this.

**Comment by heheboii11** [comment]
Yup, I got it. So here You are first defining the struct like how it looks, 
for example, you are saying that the struct ```Person```, contains a ```uint256``` and a ```string``` and those uint256 and string you are naming them as ```favouritenumber``` and ```name```. 
But in the function ```addPerson```, it is taking a parameters as a string```_name``` and a uint256 ```_favouriteNumber```. So before writing the next line of code, i.e, defining what to do with those ```_name``` and ```_favouriteNumber```, they are just the inputs for the function ```addPerson```, so in the function you are defining them as to add to the array ```listOfPeople``` as a struct```Person```. So you need to give an input for the struct ```Person``` which are ```_name``` and ```_favouriteNumber```

**Reply by yashjh** [reply]
yes
but how the compiler will get to know that which value is which one in person struct which is declared

**Reply by yashjh** [reply]
are you a blockchain engineer ?

**Reply by heheboii11** [reply]
> are you a blockchain engineer ?

Still learning to be one!!!
 And it is simple you are defining it inside the ```addPerson``` function, You are using the inputs ```_name``` and ```_favouriteNumber``` and defining that push this struct ```Person``` which has ```_name```, ```_favouriteNumber```. 
I would suggest you learn on how structs works, may be google, youtube or ask an Ai. they will help you understand more clear

**Reply by yashjh** [reply]
i also learning same so can i have your contact or handle like telegram bcz i also hace lots of doubts to aks along with regarding blockchain

**Reply by heheboii11** [reply]
Sure hit me on telegram ```@Hehebuoii```

**Reply by yashjh** [reply]
thanks

**Comment by EngrPips** [comment]
Great heads-up
